src/components/router.js

```diff
@@ -0,0 +1,47 @@
+export function Router(props, children) {
+  const match = props.actions.router.set(matcher(location.pathname, children))
```

Comment:
Should this action be called within the router module instead?  Calling it with `location.pathname` makes the view an impure function.

This was only necessary because the router module's init method is calling `actions.set({})`.  Why not call `go` instead?
```js
// In router module

    init(state, actions) {
      actions.go(window.location.pathname + window.location.search)
      addEventListener("popstate", function() {
        actions.go(window.location.pathname + window.location.search)
      })
    },
```

While we're at it, should `matcher` be called in the `go` action also?

Reply:
@speier Ping.